iml-gui/crate/src/components/grafana_chart.rs

```diff
@@ -3,29 +3,37 @@
 // license that can be found in the LICENSE file.
 
 use seed::{attrs, iframe, prelude::*};
-use serde_with::with_prefix;
-use std::collections::HashMap;
+use std::collections::BTreeMap;
 
 pub static IML_METRICS_DASHBOARD_ID: &str = "8Klek6wZz";
 pub static IML_METRICS_DASHBOARD_NAME: &str = "iml-metrics";
 
-with_prefix!(grafana_var "var-");
-
 #[derive(serde::Serialize)]
 #[serde(rename_all = "camelCase")]
 pub(crate) struct GrafanaChartData<'a> {
     pub org_id: u16,
     pub refresh: &'a str,
     pub panel_id: u16,
-    #[serde(flatten, with = "grafana_var")]
-    pub vars: HashMap<String, String>,
+    #[serde(flatten)]
+    pub vars: BTreeMap<String, String>,
 }
 
 pub(crate) fn create_chart_params<'a>(
     panel_id: u16,
     vars: impl IntoIterator<Item = (impl ToString, impl ToString)>,
 ) -> GrafanaChartData<'a> {
-    let hm: HashMap<String, String> = vars.into_iter().map(|(x, y)| (x.to_string(), y.to_string())).collect();
+    let hm: BTreeMap<String, String> = vars
+        .into_iter()
+        .map(|(x, y)| {
+            let var = x.to_string();
+            let key = match var.as_str() {
+                "to" | "from" => var.to_string(),
```

Comment:
It looks like an abuse of vars. They all are supposed to be prefixed with "var-". Why not add to and from to other parameters like panel id?

Reply:
I'm not sure why grafana made the decision to not use the `var-` prefix like everything else when using`to` and `from`, but because they are part of the query parameters I think they belong here.

Reply:
Just to follow up, all query parameters are being passed in and then serialized. If we were to take these two variables out we would have to work them into the query string outside of the serialized string. Furthermore, we would need to pass the variables individually to each function that would use it (`create_chart_params`, `performance_container` and any other function that would use it in the future). Grouping all query parameters together makes more sense to me:
```
create_chart_params(
   32,
    vec![
      ("fs_name", &model.fs_name),
      ("from", &model.mdt_usage_date_picker.from),
      ("to", &model.mdt_usage_date_picker.to)
   ]
),
```

Reply:
I mean vars and non-vars are different entities. And since we have only one way to pass them all in a query, it is decided to distinguish them by a var- prefix.

Reply:
Also any exceptions are bad, to and from are now exceptions, one will not be able to add "real" vars (to or from) easily. And correct me: any graph can be parameterized by to and from parameter, but each graph may have zero or more specific vars like fs_name.

Reply:
All charts can be parameterized by from and to, however it won't make sense for all charts (for example OST balance, where we always take latest).

In the interest of time, I'll land this and then extract the from and to params in a follow-on patch.

Reply:
Yeah, the "var-" variable are "template" variables, totally different thing from the data set variables :)